interop/klr/fromNKI.hpp

```diff
@@ -0,0 +1,511 @@
+#pragma once
+
+#include "klir_ast.hpp"
+#include "klir_common.hpp"
+#include <variant>
+#include <string>
+#include <vector>
+#include <optional>
+
+namespace Trace {
+
+struct Error {
+    std::string msg;
+    Error(std::string&& m) : msg(std::move(m)) {}
+};
+
+// Simple error handling
+template<typename T>
+using TraceResult = std::variant<T, Error>;
```

Comment:
Don't need it in this PR, but I'd love a proper Result class.
Or use c++ exceptions.